### ISSUE TITLE
AttributeError: 'cStringIO.StringO' object has no attribute 'name'

### ISSUE THREAD

**Issue by amccloud** [django-compressor/django-compressor#100]
The example CachedS3BotoStorage raises this exception when I try to do collectstatic.

I'm using compressor@develop
Django@trunk

```
Traceback (most recent call last):
  File "manage.py", line 11, in <module>
    execute_manager(settings)
  File "/Users/andrew/Labs/something/src/django/django/core/management/__init__.py", line 442, in execute_manager
    utility.execute()
  File "/Users/andrew/Labs/something/src/django/django/core/management/__init__.py", line 379, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/andrew/Labs/something/src/django/django/core/management/base.py", line 191, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "/Users/andrew/Labs/something/src/django/django/core/management/base.py", line 220, in execute
    output = self.handle(*args, **options)
  File "/Users/andrew/Labs/something/src/django/django/core/management/base.py", line 351, in handle
    return self.handle_noargs(**options)
  File "/Users/andrew/Labs/something/src/django/django/contrib/staticfiles/management/commands/collectstatic.py", line 116, in handle_noargs
    handler(path, prefixed_path, storage)
  File "/Users/andrew/Labs/something/src/django/django/contrib/staticfiles/management/commands/collectstatic.py", line 243, in copy_file
    self.storage.save(prefixed_path, source_file)
  File "/Users/andrew/Labs/something/something/../something/contrib/storages/__init__.py", line 11, in save
    self.local_storage._save(name, content)
  File "/Users/andrew/Labs/something/src/django/django/core/files/storage.py", line 198, in _save
    for chunk in content.chunks():
  File "/Users/andrew/Labs/something/src/django/django/core/files/base.py", line 65, in chunks
    counter = self.size
  File "/Users/andrew/Labs/something/src/django/django/core/files/base.py", line 39, in _get_size
    elif os.path.exists(self.file.name):
AttributeError: 'cStringIO.StringO' object has no attribute 'name'
```

The exact code for the backend is:

    from django.core.files.storage import get_storage_class
    from storages.backends.s3boto import S3BotoStorage

    class CachedS3BotoStorage(S3BotoStorage):
        def __init__(self, *args, **kwargs):
            super(CachedS3BotoStorage, self).__init__(*args, **kwargs)
            self.local_storage = get_storage_class('compressor.storage.CompressorFileStorage')()

        def save(self, name, content):
            name = super(CachedS3BotoStorage, self).save(name, content)
            self.local_storage._save(name, content)
            return name

And my settings:

    MEDIA_ROOT = os.path.join(PROJECT_DIR, 'media')
    MEDIA_URL = '//something.s3.amazonaws.com/media/'

    STATIC_ROOT = os.path.join(PROJECT_DIR, 'static')
    STATIC_URL = '//something.s3.amazonaws.com/static/'

    STATICFILES_DIRS = (
        os.path.join(PROJECT_DIR, 'static_media'),
    )

    STATICFILES_FINDERS = (
        'django.contrib.staticfiles.finders.FileSystemFinder',
        'django.contrib.staticfiles.finders.AppDirectoriesFinder',
        'compressor.finders.CompressorFinder',
    )

    DEFAULT_FILE_STORAGE = 'storages.backends.s3boto.S3BotoStorage'
    STATICFILES_STORAGE = 'something.contrib.storages.CachedS3BotoStorage'

    COMPRESS_STORAGE = STATICFILES_STORAGE
    COMPRESS_OUTPUT_DIR = 'compressed'
    COMPRESS_CSS_FILTERS = (
        'compressor.filters.css_default.CssAbsoluteFilter',
        'compressor.filters.cssmin.CSSMinFilter',
    )
    COMPRESS_JS_FILTERS = (
        'compressor.filters.jsmin.JSMinFilter',
    )

**Comment by karyon** [comment]
the issue @jezdez identified as the root cause here https://github.com/django-compressor/django-compressor/issues/100#issuecomment-5491205 was fixed in django 1.5. since 1.4 is nearing EOL, i vote for closing this.

**Comment by diox** [comment]
Agreed. We're about to remove official support for 1.4 (since it's nearing EOL) in compressor and there are enough workarounds available anyway.
